### PR TITLE
Fix morph issue on some platforms

### PR DESCRIPTION
**Pull request by shrinktofit**
Re: cocos-creator/3d-tasks#

Changelog:
 * Fix morph issue on platforms with shader >= 300 && having not float texture;
 * Fix GPU morph is failed when using rgba8 texture on some platforms.

### CODE REVIEW

File: editor/assets/chunks/morph.chunk [hunk 1]
```diff
@@ -89,7 +89,9 @@ vec3 getVec3DisplacementFromTexture(sampler2D tex, int vertexIndex) {
 #else
     vec3 result = vec3(0, 0, 0);
     for (int iTarget = 0; iTarget < CC_MORPH_TARGET_COUNT; ++iTarget) {
-        int dataPixelStart = int(fetchVec3ArrayFromTexture(tex, iTarget).r);
+        // `+0.5` because in some platforms, the "fetched integer" may have small error.
+        // For example `26` may yeild `25.99999`, which is convert to `25` instead of `26` using `int()`.
+        int dataPixelStart = int(fetchVec3ArrayFromTexture(tex, iTarget).r + 0.5);
```
Comment (shrinktofit) [inline editor/assets/chunks/morph.chunk]: @YunHsiao  Notable change

File: editor/assets/chunks/morph.chunk [hunk 2]
```diff
@@ -6,7 +6,7 @@
 #if __VERSION__ < 300
     in float a_vertexId;
     int getVertexId() {
-        return int(a_vertexId);
+        return int(a_vertexId + 0.5);
```
Comment (shrinktofit) [inline editor/assets/chunks/morph.chunk]: @YunHsiao  Notable change

File: editor/assets/chunks/morph.chunk (hunk 1 above)
Comment (YunHsiao) [reply editor/assets/chunks/morph.chunk]: 这种 offset 在 CPU 做是不是就行了，不要每个顶点都执行一遍了

File: editor/assets/chunks/morph.chunk (hunk 2 above)
Comment (YunHsiao) [reply editor/assets/chunks/morph.chunk]: 这个也一样

File: editor/assets/chunks/morph.chunk (hunk 1 above)
Comment (shrinktofit) [reply editor/assets/chunks/morph.chunk]: Fixed at https://github.com/cocos-creator/engine/pull/7268/commits/ac84a54e2640d224e6948be86ecf3a3654b6dcaa

File: editor/assets/chunks/morph.chunk (hunk 2 above)
Comment (shrinktofit) [reply editor/assets/chunks/morph.chunk]: Fixed at https://github.com/cocos-creator/engine/pull/7268/commits/ac84a54e2640d224e6948be86ecf3a3654b6dcaa

**Review by YunHsiao — APPROVED** [review]

**Pull request merged**